src/Hook/CommonHook.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright Younited
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to tech@202-ecommerce.com so we can send you a copy immediately.
+ *
+ * @author    202 ecommerce <tech@202-ecommerce.com>
+ * @copyright Younited
+ * @license   https://opensource.org/licenses/AFL-3.0  Academic Free License (AFL 3.0)
+ */
+
+namespace YounitedpayAddon\Hook;
+
+use Context;
+use Younitedpay;
+use YounitedpayClasslib\Hook\AbstractHook;
+
+class CommonHook extends AbstractHook
+{
+    const AVAILABLE_HOOKS = [
+        'actionFrontControllerSetMedia',
+        'actionAdminControllerSetMedia',
+    ];
+
+    public function actionAdminControllerSetMedia($params)
+    {
+        /** @var \AdminController $controller */
+        $controller = Context::getContext()->controller;
+        if ($controller->controller_name === 'AdminOrders') {
+            $controller->addJS(_PS_MODULE_DIR_ . $this->module->name . '/views/js/admin.js');
+            \Media::addJsDef([
+                'younitedpay' => [
+                    'translations' => [
+                        'slip_refund' => $this->l('Generate a Credit Slip must be selected to refund with Younited Pay'),
```

Comment:
`$this->l('Generate a Credit Slip must be selected to refund with Younited Pay', 'CommonHook'),` instead ?

Reply:
As there is our TranslateTrait class (from ClassLib) used here 
there is no need to specify the concerned className.

Tested and it works in french translation.